packages/dd-trace/src/log.js

```diff
@@ -7,9 +7,22 @@ const _default = {
   error: err => console.error(err) /* eslint-disable-line no-console */
 }
 
+const _logLevels = ['trace', 'debug', 'info', 'warn', 'error']
+
 let _logger
 let _enabled
 let _deprecate
+let _logLevel
+
+const _isLogLevelEnabled = (level) => {
+  return !_logLevel || _logLevels.indexOf(level) >= _logLevels.indexOf(_logLevel)
```

Comment:
This is currently too heavy to be called for every log entry. Can numerical values be stored instead? This would make the comparison a lot more lightweight.

Reply:
Makes sense, that wasn't something I'd really considered. I wanted to make sure the user could just input a string for their settings, and then also that this function accepts a string, so that the code remains readable and the user or future work here doesn't have to worry about keeping track of the mappings between the log level and their numerical value in multiple places.

So, to address this i've updated `_isLogLevelEnabled` and `_setLogLevel` along with `_logLevels`, so that `_logLevels` is just an object of key/value pairs `debug: 0, error: 1, etc etc`.  This way in the updated code the call to `indexOf` is removed completely and it's just a single object lookup for the log level key in the `_logLevels` object, then a numerical comparison of that value.

Additionally for the case of an invalid value i've just added a `_defaultLogLevel` that i've set to `debug` since this matches the current behavior.

All in all it's become a bit more complicated than i'd like here but i believe this allows for things to be both performant on the comparison, and also easy to use/extensible for any future work here. Lmk what you think